src/tools/ceph_dedup_tool.cc

```diff
@@ -570,21 +583,31 @@ class SampleDedupWorkerThread : public Thread
       return cur_reference >= dedup_threshold && dedup_threshold != -1;
     }
 
-    FpStore(size_t chunk_threshold) : dedup_threshold(chunk_threshold) { }
+    FpStore(size_t chunk_threshold, uint32_t report_period) :
+      dedup_threshold(chunk_threshold), report_period(report_period) {
+      start = ceph_clock_now();
+      next_report = start;
+      next_report += report_period;
+    }
 
   private:
     ssize_t dedup_threshold = -1;
     std::unordered_map<std::string, dup_count_t> fp_map;
     std::shared_mutex fingerprint_lock;
+    utime_t start;
+    utime_t next_report;
+    uint32_t report_period = default_report_period;
```

Comment:
const uint32_t report_period -- can you express the default where the option is defined?

Reply:
Moved to the options_description. Please see the last commit.